cmd/kubectl-gadget/utils/trace.go

```diff
@@ -371,7 +371,15 @@ func getTraceListFromID(traceID string) (*gadgetv1alpha1.TraceList, error) {
 // SetTraceOperation sets the operation of an existing trace.
 // If trace does not exist an error is returned.
 func SetTraceOperation(traceID string, operation string) error {
-	traces, err := getTraceListFromID(traceID)
+	// We have to wait for the end of previous operation before changing the trace
+	// operation.
+	// Trace controller set Annotations to nil when it is dealing with an
+	// operation.
+	// Thus, we need to wait for Annotations to become nil (empty) before being
+	// able to change the current operation.
+	// It is the same like when you are in the restaurant, you need to wait the
+	// chef to cook the main dishes before ordering the dessert.
```

Comment:
```suggestion
	// We have to wait for the previous operation to finish before changing the trace
	// operation.
	// The trace controller sets Annotations to nil once it has dealt with an operation.
	// Thus, to avoid losing operations, we need to wait for the Annotations field to 
	// become nil (empty) before changing to the current operation.
	// It is the same like when you are in the restaurant, you need to wait for the
	// chef to cook the main dishes before ordering the dessert.
```